Workers/DatabaseSyncWorker/DatabaseChanges.cs

```diff
@@ -14,12 +14,13 @@ public class DatabaseChanges<TType> : IDisposable
     {
         private ImmutableArray<TType> changes = ImmutableArray<TType>.Empty;
         private CancellationTokenSource tcs;
+        private readonly Task task;
 
         public DatabaseChanges(PostgresOptions postgresOptions, string tableName)
         {
             tcs = new CancellationTokenSource();
 
-            Task.Factory.StartNew(async unusedStateObject =>
+            task = Task.Factory.StartNew(async unusedStateObject =>
```

Comment:
`Task.Factory.StartNew` has an overload that takes a cancellation token, do we want to pass that in there too? 

I'm not sure what the behaviour is having the same token on a "child" and "parent" task.